helix-core/src/test/java/org/apache/helix/model/TestClusterTopologyConfig.java

```diff
@@ -0,0 +1,84 @@
+package org.apache.helix.model;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.util.Iterator;
+import org.apache.helix.HelixException;
+import org.apache.helix.controller.rebalancer.topology.Topology;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+
+public class TestClusterTopologyConfig {
+
+  @Test
+  public void testClusterNonTopologyAware() {
+    ClusterConfig testConfig = new ClusterConfig("testId");
+    testConfig.setTopologyAwareEnabled(false);
+    ClusterTopologyConfig clusterTopologyConfig = ClusterTopologyConfig.fromClusterConfig(testConfig);
+    Assert.assertEquals(clusterTopologyConfig.getEndNodeType(), Topology.Types.INSTANCE.name());
+    Assert.assertEquals(clusterTopologyConfig.getFaultZoneType(), Topology.Types.INSTANCE.name());
```

Comment:
Is this correct? Why the fault zone type is INSTANCE name? shall we change the default value to be meaningful?

Reply:
This is what I read https://github.com/apache/helix/pull/1945/files#diff-3bfc7aaa9b4578cd05434a4b4564dd61aa561f340ba7c08d1395d3f50a881fc2L238-L243, so I just keep the behavior. 
Since I'm not really sure on the impact of changing this default value, I prefer keeping it the same. 
@xyuanlu any input? I saw you introduced [this](https://github.com/apache/helix/commit/0869e729530d83ba55228b2d44a79ae44175c081).

Reply:
@qqu0127 I agree that @xyuanlu should take a look. Could anyone of you check if `testCreateClusterTopologyWithDefaultTopology` is correctly testing this case, by the way? On the first glance that should have caught this case? Thanks.

Reply:
TestTopology.testCreateClusterTopologyWithDefaultTopology PASS with the patch.

Reply:
The default fault zone type and end node type is INSTANCE when topologyAware is not enabled.  When it is enabled, the default fault zone type is ZONE and end node type is INSTANCE. 
This behavior is added in 2017 by change https://github.com/apache/helix/commit/cb024650c25c6ddee7348d37fb67fa1a2e93bff1#

IMHO, the current behavior make sense. When no topology enabled, the structure should be flat, meaning zone and end node should be the same.